internal/pkg/service/appsproxy/proxy/testutil/app.go

```diff
@@ -40,6 +40,27 @@ func StartAppServer(t *testing.T) *AppServer {
 		assert.NoError(t, c.Close(websocket.StatusNormalClosure, ""))
 	})
 
+	mux.HandleFunc("/ws2", func(w http.ResponseWriter, r *http.Request) {
+		c, err := websocket.Accept(w, r, nil)
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithTimeout(r.Context(), time.Second*10)
```

Comment:
I understand that it is not easy to rework the test so you do not need to use timer but I would decrease the timeout
```suggestion
		ctx, cancel := context.WithTimeout(r.Context(), 600 * time.Millisecond)
```
And swap the `time.Second/Millisecond` definition to end as it is in other cases